src/scraper/acg-rip.ts

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2023 IROHA LAB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Axios from 'axios';
+import { inject, injectable } from 'inversify';
+import { Item } from '../entity/Item';
+import { ItemType } from '../entity/item-type';
+import { Team } from '../entity/Team';
+import { Publisher } from '../entity/publisher';
+import { TaskOrchestra } from '../task/task-orchestra';
+import { ConfigLoader, ItemStorage, TYPES } from '../types';
+import { logger } from '../utils/logger-factory';
+import { captureException } from '../utils/sentry';
+import { getMediaFiles } from '../utils/torrent-utils';
+import { BaseScraper } from './abstract/base-scraper';
+import cheerio = require('cheerio');
+
+@injectable()
+export class AcgRipScraper extends BaseScraper<number> {
+    private static _host = 'https://acg.rip';
+        
+    constructor(
+        @inject(TYPES.ItemStorage) store: ItemStorage<number>,
+        @inject(TaskOrchestra) taskOrchestra: TaskOrchestra,
+        @inject(TYPES.ConfigLoader) config: ConfigLoader
+    ) {
+        super(taskOrchestra, config, store);
+    }
+
+    public async executeMainTask(pageNo?: number): Promise<{ items: Item<number>[]; hasNext: boolean; }> {
+        try {
+            let listPageUrl = AcgRipScraper._host;
+            if (pageNo) {
+                listPageUrl += `/page/${pageNo}`;
+            }
+            logger.info('execute_main_task', { pageNo });
+            const resp = await Axios.get(listPageUrl);
+
+            const $ = cheerio.load(resp.data);
+            const trList = Array.from($('table.post-index > tbody > tr'));
+            let items: Array<Item<number>> = [];
+            trList.forEach(tr => {
+                const item = new Item<number>();
+
+                // type
+                item.type = new ItemType<number>();
+                item.type.id = 1;  // hard coded type 1 as anime
+                item.type.name = '动画';
```

Comment:
Where should I get item type while all scraped items are anime.

Reply:
<img width="569" alt="chrome_2023-04-06-195053-1879" src="https://user-images.githubusercontent.com/55684557/230355999-939b4584-82c6-4221-90f0-f2501d79d83c.png">
I think should scrap every tab each time executing MainTask

Reply:
The current scraper architecture does not support multiple entry points and would require a large refactoring to do so. For acg.rip, we may only need `动画` and `日剧`. It may be more appropriate add a different mode which is inherited from AcgRipBaseScraper.

Reply:
You can scrap every category in one shot, and mix these 6 list into one then filter and store to get IDs, the rest should the same, when there is a need for scrap the next page, just add page number to each category

Reply:
The frequency of updates varies for each type, and the `hasNext` value may differ for each category, so there may be a situation where one requires scraping next page while the others do not.

Reply:
It doesn't matter which one requires hasNext and which one does not, because every link on all these pages is unique, you can combine these 9 categories into one list, then if the combined list has at least 1 item not scrapped then considering it hasNext=true, and go to next page for all these 9 categories. repeat the previous step.